lib/phoenix/live_dashboard/pages/metrics_page.ex

```diff
@@ -63,36 +61,107 @@ defmodule Phoenix.LiveDashboard.MetricsPage do
       <:item :for={item <- @items} name={item} label={format_nav_name(item)} method="redirect">
         <div :if={@metrics} class="phx-dashboard-metrics-grid row">
           <%= for {metric, id} <- @metrics do %>
-            <%= live_component ChartComponent, id: id(id, @nav), metric: metric %>
+            <.live_metric_chart id={id(id, @nav)} metric={metric} />
           <% end %>
         </div>
       </:item>
     </.live_nav_bar>
     """
   end
 
+  @doc false
+  attr :id, :string,
+    required: true,
+    doc: "Because is a stateful `Phoenix.LiveComponent` an unique id is needed."
+
+  attr :metric, :any, required: true, doc: "Metric to be represented in the chart"
+
+  def live_metric_chart(%{id: id, metric: metric}) do
+    assigns = assigns_from_metric(id, metric)
+
+    ~H"""
+    <Phoenix.LiveDashboard.PageBuilder.live_chart {assigns} />
+    """
+  end
```

Comment:
I also wanted to move this to `PageBuilder` with `@doc false`. However, I couldn't test a functional component that renders a live component. The interesting part of this component is the "magic" in `assigns_from_metric`. I wanted to test the component using the compile time validations and default values, but I think it is impossible. If I use `rendered_to_string` function I have the following error:

```elixir
    assigns = %{}

    rendered_to_string(~H"""
      <.live_chart id={1} kind={:last_value} title="Title" />
    """)
```

![image](https://user-images.githubusercontent.com/1745859/223288682-80626f0f-83d6-4a19-9ac6-aaf9da024ffe.png)

If I use `render_component/2`, then I'm not using the function component's validations and default values. (If I have time, I'll try to make a PR for https://github.com/phoenixframework/phoenix_live_view/issues/2407 )

For this reason, I opted to move this function back to metrics_page, make public the `assigns_from_metric` function and make it tests. Not ideal, but it is the best thing I could think of.